src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -3,6 +3,7 @@
 import bio.terra.workspace.common.exception.DuplicateWorkspaceException;
 import bio.terra.workspace.common.exception.WorkspaceNotFoundException;
 import bio.terra.workspace.generated.model.WorkspaceDescription;
+import bio.terra.workspace.generated.model.WorkspaceStageEnum;
```

Comment:
I'm still not wild about passing model classes down to the WorkspaceDao. This locks us into any changes to our representation of the workspaces internally being reflected in our public API, or vice versa. I think the cost of writing a conversion function to go `api model` <-> `internal enum representation` is tiny. I think the cost of this future proofing is low, and linking API & DB layers is a classic painful problem.
Examples:
- Something that might be a public enum might later have more state. 
- We might want to rename enums in the API. Renaming enums in the DB require DB migrations.

Reply:
Sounds good. It's a pretty minor difference to implement now, so if there's some expected benefit in the future that seems worthwhile. I haven't used the pattern of separate API model / internal representation before, so let me know if there are style improvements on where things should live, where the transform should happen, etc

Reply:
All the conversion between the request/response models to the internal representation should happen at the request/response layer, i.e. the controllers. The serialization layers at the top/bottom of the service should be in charge of converting the internal representation to the serialized representation. The services/flights/DAOs should communicate with the internal representation. (The DAO handles an equivalent conversion to its own representation, the SQL queries/params/results.)

Reply:
That seems like a reasonable pattern, but at least at the moment a number of service methods directly take the input request. Refactoring those out seems like a larger effort